web/src/integration_url_modal.js

```diff
@@ -57,25 +77,53 @@ export function show_generate_integration_url_modal(api_key) {
             update_url();
         });
 
-        function update_url() {
+        function update_url(update_events = false) {
             selected_integration = integration_input_dropdown_widget.value();
             if (selected_integration === default_integration_option.unique_id) {
                 $integration_url.text(default_url_message);
                 $dialog_submit_button.prop("disabled", true);
                 return;
             }
 
+            const params = new URLSearchParams({api_key});
+
+            const selected_integration_data = realm.realm_incoming_webhook_bots.find(
+                (bot) => bot.name === selected_integration,
+            );
+            const all_event_types = selected_integration_data.all_event_types;
+
+            if (
+                all_event_types !== null &&
+                !update_events &&
+                $show_integration_events.prop("checked")
+            ) {
+                all_event_types.sort((a, b) => a.localeCompare(b));
+                const events = render_integration_events({
+                    events: all_event_types,
+                });
+                $("#integration-events-container").empty().append(events);
+            } else if (!update_events && all_event_types === null) {
+                $("#integration-events-toggle-wrapper").addClass("hide");
+                $("#integration-events-ui").addClass("hide");
+                $("#integration-events-container").empty();
+                $(".integration-event-checkbox").prop("checked", false);
```

Comment:
For more specificity here, whenever referencing this class I'd also reference the container id (see my renaming comments). So this would be:

`$("#integration-events-container .integration-event").prop("checked", false);`